src/overlay025/poketch_system.c

```diff
@@ -0,0 +1,790 @@
+    {POKETCH_APPID_DOWSINGMACHINE,      FS_OVERLAY_ID(overlay34)},
+    {POKETCH_APPID_COUNTER,             FS_OVERLAY_ID(overlay35)},
+    {POKETCH_APPID_PEDOMETER,           FS_OVERLAY_ID(overlay36)},
+    {POKETCH_APPID_DAYCARECHECKER,      FS_OVERLAY_ID(overlay40)},
+    {POKETCH_APPID_ROULETTE,            FS_OVERLAY_ID(overlay41)},
+    {POKETCH_APPID_COINTOSS,            FS_OVERLAY_ID(overlay42)},
+    {POKETCH_APPID_MOVETESTER,          FS_OVERLAY_ID(overlay43)},
+    {POKETCH_APPID_MATCHUPCHECKER,      FS_OVERLAY_ID(overlay44)},
+    {POKETCH_APPID_ALARMCLOCK,          FS_OVERLAY_ID(overlay45)},
+    {POKETCH_APPID_KITCHENTIMER,        FS_OVERLAY_ID(overlay46)},
+    {POKETCH_APPID_MARKINGMAP,          FS_OVERLAY_ID(overlay47)},
+    {POKETCH_APPID_BERRYSEARCHER,       FS_OVERLAY_ID(overlay48)},
+    {POKETCH_APPID_COLORCHANGER,        FS_OVERLAY_ID(overlay49)},
+    {POKETCH_APPID_CALENDAR,            FS_OVERLAY_ID(overlay50)},
+    {POKETCH_APPID_LINKSEARCHER,        FS_OVERLAY_ID(overlay52)},
+    {POKETCH_APPID_RADARCHAINCOUNTER,   FS_OVERLAY_ID(overlay53)},
+    {POKETCH_APPID_POKEMONHISTORY,      FS_OVERLAY_ID(overlay54)}
+};
+
+static PoketchSystem * ov25_02253CE0 (void)
+{
+    return sub_0203D158();
+}
+
+
+void PoketchSystem_Create(FieldSystem *fieldSys, PoketchSystem **poketchSys, SaveData *saveData, BGL *param3, NNSG2dOamManagerInstance *param4)
+{
+    PoketchSystem *new_system;
+
+    Heap_Create(3, 7, 49152);
+    Heap_Create(3, 8, 49152);
+
+    new_system = Heap_AllocFromHeap(7, sizeof(PoketchSystem));
+
+    if (new_system != NULL) {
+        *poketchSys = new_system;
+
+        new_system->fieldSys = fieldSys;
+        new_system->saveData = saveData;
+        new_system->poketchData = SaveData_PoketchData(saveData);
+        new_system->bgl = param3;
+        new_system->unk_50 = param4;
+
+        if (PoketchSystem_InitInternal(new_system)) {
+            sub_0201E3D8();
+            sub_0201E450(4);
+
+            new_system->poketchSysPtr = poketchSys;
+            new_system->unk_38 = SysTask_Start(ov25_02253E94, new_system, 4);
+
+            SysTask_Start(PoketchSystem_MainTask, new_system, 0);
+        }
+    }
+}
+
+void ov25_02253D5C (PoketchSystem *poketchSys)
+{
+    if ((poketchSys->unk_00 != 3) && (poketchSys->unk_00 != 4)) {
+        ov25_02253E9C(poketchSys, 3);
+    }
+}
+
+BOOL ov25_02253D70 (PoketchSystem *poketchSys)
+{
+    return poketchSys == NULL;
+}
+
+void ov25_02253D7C (PoketchSystem *poketchSys, int param1, u32 param2)
+{
+    switch (param1) {
+    case 0:
+        break;
+    case 1:
+        poketchSys->unk_04 = 1;
+        break;
+    case 5:
+    {
+        u32 step_count = PoketchData_StepCount(poketchSys->poketchData);
+
+        if (++step_count > 99999) {
+            step_count = 0;
+        }
+
+        PoketchData_SetStepCount(poketchSys->poketchData, step_count);
+        poketchSys->pedometerUpdated = TRUE;
+    }
+    break;
```

Comment:
Is the additional block needed? Can this `u32` be declared outside the switch block to remove the additional scope block?

Reply:
Moving the u32 causes the checksum to fail sadly.

Reply:
Unfortunate. Could you reorganize the block a bit, then, like so?

```c
    case 5: {
        u32 step_count = PoketchData_StepCount(poketchSys->poketchData);

        if (++step_count > 99999) {
            step_count = 0;
        }

        PoketchData_SetStepCount(poketchSys->poketchData, step_count);
        poketchSys->pedometerUpdated = TRUE;

        break;
    }
```